src/util/timer.h

```diff
@@ -52,61 +57,49 @@ class SuspendableTimer : public Timer {
 
 class ScopedTimer {
   public:
-    ScopedTimer(const char* key, int i,
-                Stat::ComputeFlags compute = kDefaultComputeFlags)
-            : m_pTimer(NULL),
-              m_cancel(false) {
-        if (CmdlineArgs::Instance().getDeveloper()) {
-            initialize(QString(key), QString::number(i), compute);
+    // Allows the timer to contain a format string which is only assembled
+    // when we're not in `--developer` mode.
+    /// @param compute Flags to use for the Stat::ComputeFlags (can be omitted)
+    /// @param key The format string for the key
+    /// @param args The arguments to pass to the format string
+    template<typename T, typename... Ts>
+    ScopedTimer(Stat::ComputeFlags compute, T&& key, Ts&&... args)
+            : m_timer(std::nullopt) {
+        static_assert(!std::is_same_v<std::decay_t<T>, char const*>,
+                "string type likely not UTF-16, please pass QStringLiteral by "
+                "means of u\"key text\"_s");
```

Comment:
```suggestion
                "Passing char[] is not supported. Wrap it via QStringLiteral() "
                "to avoid runtime UTF-16 conversion.");
```